### PR TITLE
Add manual bot restart feature and clean up hardcoded Telegram ID

### PR DESCRIPTION
**Pull request by d8rt8v**
**Add manual bot restart feature and clean up hardcoded Telegram ID**

- Implement manual restart functionality for the bot via Telegram
- Automatically restart the bot upon encountering errors
- Remove hardcoded Telegram ID from the script for flexibility

Nice bot btw!

### CODE REVIEW

File: clicker.py [hunk 1]
```diff
@@ -486,11 +492,16 @@ async def answer(event):
...
-    elif user_id == 6583452530 and 'balance' in db and db['balance']:
+    elif 'balance' in db and db['balance']:
         db['balance'] = False
         b = text.split('Balance: ')[1].split('\n')[0]
         await client.send_message(admin, f'💡 Balance: {b}💛')
 
+    elif text == '/restart':
+        await _sendMessage('Restarting')
+        print('[!] Restarting')
+        os.execv(sys.executable, ['python3', script_path] + sys.argv)
```
Comment (IgorGradov) [inline clicker.py:503]: Can be python and python3.11 and python 3.12 as well

**Comment by d8rt8v** [comment]
@Poryaei can you review my PR?

**Comment by IgorGradov** [comment]
You've just removed some of the functionality.....

File: clicker.py [hunk 2]
```diff
@@ -399,8 +406,6 @@ async def answer(event):
     text = event.raw_text
     user_id = event.sender_id
     
-    if not user_id in [admin, 6583452530]:
```
Comment (IgorGradov) [inline clicker.py:402]: By removing this line, you allow anyone to control your bot.

File: clicker.py [hunk 3]
```diff
@@ -486,11 +492,16 @@ async def answer(event):
         await _sendMessage('👋')
         sys.exit()
   
-    elif user_id == 6583452530 and 'balance' in db and db['balance']:
```
Comment (IgorGradov) [inline clicker.py:489]: you removed the balance parsing here

File: clicker.py (hunk 1 above)
Comment (IgorGradov) [reply clicker.py:503]: there is no guarantee that the right python interpreter will be called as python3

File: clicker.py [hunk 4]
```diff
@@ -376,6 +378,11 @@ def startMin(self):
                 print('[~] Generating New Auth')
                 time.sleep(random.randint(2, 4))
                 self.webAppData = self.generateAuthToken()
+                if error_start_time is None:
+                    error_start_time = time.time()
```
Comment (IgorGradov) [inline clicker.py:382]: you don't reset the timer if he manages keep mining

File: clicker.py [hunk 5]
```diff
@@ -376,6 +378,11 @@ def startMin(self):
...
+                elif time.time() - error_start_time > 120:
+                    print(f'[!] The bot was restarted due to persistent errors')
+                    os.execv(sys.executable, ['python3', script_path] + sys.argv)
```
Comment (IgorGradov) [inline clicker.py:385]: duplicated line with line 503

**Review by IgorGradov — CHANGES_REQUESTED** [review]

**Comment by Poryaei** [comment]
Thank you for the effort you put in!
But the code you wrote has some issues! and it causes disruption, as our friend @IgorGradov mentioned.
Unfortunately, the pull request is not being accepted.

File: clicker.py (hunk 2 above)
Comment (Poryaei) [reply clicker.py:402]: 6583452530

It's @notcoin_bot user id

File: clicker.py (hunk 1 above)
Comment (Poryaei) [inline clicker.py:503]: I think this restart will run into trouble! Before that, the bot should be stopped
await client.disconnect()

#clicker ... stop

**Pull request closed without merging**